ocean_provider/routes/consume.py

```diff
@@ -93,8 +93,22 @@ def fileinfo():
 
     if did:
         asset = get_asset_from_metadatastore(get_metadata_url(), did)
+        if not asset:
+            return error_response(
```

Comment:
can we add these errors to the validators instead? I think we can condition on the presence of the "did" field and retrieve asset and service, i.e. a custom validation rule like `did:exists_in_store`. This will avoid repetition between here and initialize and offer a more uniform error handling.

Reply:
I would not like to resolve same asset for maybe several times -> wasted CPU, RAM, bandwidth, extra aqua load, etc.   We just need to handle the case where did does not exists or serviceId does not exists